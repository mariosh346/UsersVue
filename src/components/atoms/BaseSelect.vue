<template>
  <q-select
    :model-value="selectedOption"
    :options="options"
    :label="label"
    filled
    v-bind="$attrs"
    @update:model-value="$emit('update:modelValue', $event.value)"
  />
</template>

<script setup lang="ts">
import { computed } from 'vue';
import type { SelectOption } from 'src/types/forms';
import type { QSelectProps } from 'quasar';

const props = defineProps<QSelectProps & {
  modelValue: string;
  options: Array<SelectOption>;
}>();

const selectedOption = computed(() => props.options && props.options.find((option) => option.value === props.modelValue));

defineEmits<{
  (e: 'update:modelValue', value: typeof props["modelValue"]): void;
}>();
</script>
