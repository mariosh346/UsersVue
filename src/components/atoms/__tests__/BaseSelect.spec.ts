import { describe, it, expect, beforeEach } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import BaseSelect from "../BaseSelect.vue";

describe('BaseSelect', () => {
  const options = [{ value: 'a', label: 'Option A' }, { value: 'b', label: 'Option B' }];
  let wrapper: ReturnType<typeof shallowMount>;
  const createWrapper = () => shallowMount(BaseSelect, {
    props: {
      modelValue: 'a',
      options,
      label: 'Test Label'
    },
    global: {
      stubs: ['q-select']
    }
  });

  beforeEach(() => {
    wrapper = createWrapper();
  });

  it('renders and selects correct option', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
