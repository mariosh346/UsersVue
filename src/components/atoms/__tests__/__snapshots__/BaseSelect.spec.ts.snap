// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseSelect > matches snapshot 1`] = `"<q-select-stub modelvalue="[object Object]" label="Test Label" virtualscrollhorizontal="false" virtualscrollslicesize="10" virtualscrollsliceratiobefore="1" virtualscrollsliceratioafter="1" virtualscrollstickysizestart="0" virtualscrollstickysizeend="0" noerroricon="false" reactiverules="false" lazyrules="false" stacklabel="false" hidehint="false" filled="true" outlined="false" borderless="false" standout="false" square="false" loading="false" labelslot="false" bottomslots="false" hidebottomspace="false" rounded="false" dense="false" itemaligned="false" counter="false" clearable="false" disable="false" autofocus="false" multiple="false" displayvaluehtml="false" options="[object Object],[object Object]" hideselected="false" hidedropdownicon="false" fillinput="false" optionsdense="false" optionshtml="false" optionscover="false" menushrink="false" popupnoroutedismiss="false" useinput="false" usechips="false" mapoptions="false" emitvalue="false" disabletabselection="false" inputdebounce="500" tabindex="0" behavior="default"></q-select-stub>"`;
